alg/viewshed.cpp

```diff
@@ -282,29 +282,39 @@ GDALDatasetH GDALViewshedGenerate(
     }
 
     /* calculate the area of interest */
+    constexpr double EPSILON = 1e-8;
     int nXStart =
         dfMaxDistance > 0
-            ? (std::max)(0, static_cast<int>(std::floor(
-                                nX - adfInvGeoTransform[1] * dfMaxDistance)))
+            ? (std::max)(
```

Comment:
just curios why the `(std::max)` instead of `std::max` ?

Reply:
I've added a commit to remove those parentheses

Reply:
> just curios why the `(std::max)` instead of `std::max` ?

Given include file ordering scenarios, it is my experience that we don't always get the behavior we hope for with NOMINMAX. For PDAL, especially when using `std::max` and `std::min` in header files, we always enclose the functions so we are never beholden to some external definition configuration of the NOMINMAX macro.

Reply:
> Given include file ordering scenarios, it is my experience that we don't always get the behavior we hope for with NOMINMAX

``add_definitions(-DNOMINMAX)`` does the job nicely, at least for .cpp compilation units. But your point may be valid for public headers where you can't control easily that. As far as I can see, we only have one public header "gdalcachedpixelaccessor.h" with std::min use in it (but it is rather esoteric, so I don't think anyone outside of GDAL use it)